src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,48 +1,34 @@
 // (C) 2020 GoodData Corporation
 import * as React from "react";
 import without = require("lodash/without");
-import { Execution } from "@gooddata/typings";
-import { stringToFloat } from "../../../helpers/utils";
 import { IntlWrapper } from "../../core/base/IntlWrapper";
 import { IntlTranslationsProvider, ITranslationsComponentProps } from "../../core/base/TranslationsProvider";
 import PushpinSizeLegend from "./legends/PushpinSizeLegend";
-import { getGeoData, getFormatFromExecutionResponse } from "../../../helpers/geoChart";
-import { IGeoConfig, IGeoData } from "../../../interfaces/GeoChart";
+import { IGeoConfig, IGeoData, IPushpinCategoryLegendItem } from "../../../interfaces/GeoChart";
 import { TOP } from "../../visualizations/chart/legend/PositionTypes";
-import { isTwoDimensionsData } from "../../../helpers/executionResultHelper";
+import PushpinCategoryLegend, {
+    IPushpinCategoryLegendProps,
+} from "../geoChart/legends/PushpinCategoryLegend";
 
 export interface IGeoChartLegendRendererProps {
     config: IGeoConfig;
-    execution: Execution.IExecutionResponses;
+    geoData: IGeoData;
+    sizeData?: number[];
+    segmentByData?: IPushpinCategoryLegendItem[];
+    sizeFormat?: string;
     locale: string;
     position?: string;
+    onItemClick?(item: IPushpinCategoryLegendItem): void;
 }
+
 export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
-    if (!props.execution) {
-        return null;
-    }
-    const {
-        execution: { executionResult, executionResponse },
-        config: { mdObject: { buckets = [] } = {} },
-        locale,
-        position = TOP,
-    } = props;
-    const geoData: IGeoData = getGeoData(buckets, executionResponse.dimensions);
-    const { data } = executionResult;
-    const { size } = geoData;
-    const classes = `geo-legend s-geo-legend position-${position}`;
+    const { locale, position = TOP, segmentByData, sizeData, sizeFormat } = props;
 
-    if (!size || !isTwoDimensionsData(data)) {
-        return null;
-    }
+    const classes = `geo-legend s-geo-legend position-${position}`;
     return (
         <div className={classes}>
-            {size &&
-                renderPushpinSizeLegend(
-                    data[size.index].map(stringToFloat),
-                    getFormatFromExecutionResponse(size.index, executionResponse),
-                    locale,
-                )}
+            {segmentByData && renderPushpinCategoryLegend(props)}
```

Comment:
we should only pass what we need for  category legend

Reply:
If we pass only what need for Category Legend, the function will have greater than 4 parameters. I think it is not good.